frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-install.tsx

```diff
@@ -0,0 +1,304 @@
+import { ConfigMapModel, NodeModel, ClusterServiceVersionModel } from '@console/internal/models';
+import { OCSModel } from '../../models';
+import { ClusterServiceVersionKind } from '@console/internal/components/operator-lifecycle-manager/index';
+import { CreateYAML } from '@console/internal/components/create-yaml';
+
+import './ocs-install.scss';
+
+const tableColumnClasses = [
+  classNames('col-md-4', 'col-sm-5', 'col-xs-8'),
+  classNames('col-md-1', 'hidden-sm', 'hidden-xs'),
+  classNames('col-md-1', 'col-sm-3', 'hidden-xs'),
+  classNames('col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-md-2', 'col-sm-4', 'col-xs-4'),
+];
+
+const NodeTableHeader = () => {
+  return [
+    {
+      title: 'Name', sortField: 'metadata.name', transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Role',
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'CPU',
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Memory',
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: 'Capacity',
+      props: { className: tableColumnClasses[4] },
+    },
+    {
+      title: 'Devices',
+      props: { className: tableColumnClasses[5] },
+    },
+  ];
+};
+NodeTableHeader.displayName = 'NodeTableHeader';
+const configMapsForAllNodes = {};
+
+const getConfigMaps = () => {
+  k8sGet(ConfigMapModel, null, 'openshift-storage').then(configMaps => {
+    console.log(configMaps, 'configmaps')
+    configMaps.items.forEach(config => {
+      const nodeName = config.metadata.labels && config.metadata.labels['rook.io/node'];
+
+      if (typeof nodeName !== 'undefined') {
+        configMapsForAllNodes[nodeName] = (JSON.parse(config.data.devices)).length;
+      }
+
+    });
+
+    console.log(configMapsForAllNodes, 'maps');
+  });
+};
+
+//need to use this for selection table
+// const NodeTableRows = (componentProps) => {
+//   return _.map(componentProps.data, obj => {
+//     if (obj && obj.metadata) {
+//       const cells = NodeTableRow(obj);
+//       const uid = obj.metadata.uid;
+//       const selected = false;
+//       return {
+//         selected,
+//         cells,
+//         uid,
+//       };
+//     }
+//   });
+// };
+
+const NodeTableRow: React.FC<NodeTableRowProps> = ({ obj: node, index, key, style }) => {
+  const roles = getNodeRoles(node).sort();
+  const devicesCount = configMapsForAllNodes[node.metadata.name] || 0;
+
+  return (
+    <TableRow id={node.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={tableColumnClasses[0]}>
+        <ResourceLink kind="Node" name={node.metadata.name} title={node.metadata.uid} />
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>
+        {roles.length ? roles.join(', ') : '-'}
```

Comment:
I see this code also exists in `public/components/node.tsx`.

I'd prefer not to rely on "zero is falsy" semantics here:

```tsx
{roles.length > 0 ? roles.join(', ') : '-'}
```

or even shorter:

```jsx
{roles.join(', ') || '-'}
```

but when you decide to change it here, please also change it in `public/components/node.tsx`.